### PR TITLE
TEZ-4564: TezClient to expose Tez AM host:port

### PR DESCRIPTION
**Pull request by abstractdog**
This PR exposes Tez AM host and port from TezClient (and the underlying FrameworkClient).

### CODE REVIEW

**Comment by tez-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  26m 35s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   5m 44s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  12m 13s |  master passed  |
| +1 :green_heart: |  compile  |   1m 19s |  master passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1  |
| +1 :green_heart: |  compile  |   1m  9s |  master passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06  |
| +1 :green_heart: |  checkstyle  |   1m 23s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m 11s |  master passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1  |
| +1 :green_heart: |  javadoc  |   0m 59s |  master passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06  |
| +0 :ok: |  spotbugs  |   1m 20s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   2m 56s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 50s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 55s |  the patch passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1  |
| +1 :green_heart: |  javac  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 46s |  the patch passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06  |
| +1 :green_heart: |  javac  |   0m 46s |  the patch passed  |
| -0 :warning: |  checkstyle  |   0m 13s |  tez-api: The patch generated 2 new + 32 unchanged - 0 fixed = 34 total (was 32)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 34s |  the patch passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1  |
| +1 :green_heart: |  javadoc  |   0m 35s |  the patch passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06  |
| +1 :green_heart: |  findbugs  |   2m 16s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 16s |  tez-api in the patch passed.  |
| +1 :green_heart: |  unit  |   4m 59s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 24s |  The patch does not generate ASF License warnings.  |
|  |   |  69m  0s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.44 ServerAPI=1.44 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-356/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/356 |
| JIRA Issue | TEZ-4564 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux fa043c3c77c2 5.15.0-94-generic #104-Ubuntu SMP Tue Jan 9 15:25:40 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / a1fcddb8b |
| Default Java | Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06 |
| checkstyle | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-356/1/artifact/out/diff-checkstyle-tez-api.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-356/1/testReport/ |
| Max. process+thread count | 376 (vs. ulimit of 5500) |
| modules | C: tez-api tez-dag U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-356/1/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by tez-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 33s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   6m  8s |  Maven dependency ordering for branch  |
| -1 :x: |  mvninstall  |  13m 35s |  root in master failed.  |
| +1 :green_heart: |  compile  |   1m  5s |  master passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1  |
| +1 :green_heart: |  compile  |   0m 53s |  master passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06  |
| +1 :green_heart: |  checkstyle  |   1m  1s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m 33s |  master passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1  |
| +1 :green_heart: |  javadoc  |   0m 40s |  master passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06  |
| +0 :ok: |  spotbugs  |   1m 17s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   2m 51s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 56s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  1s |  the patch passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1  |
| +1 :green_heart: |  javac  |   1m  1s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 51s |  the patch passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06  |
| +1 :green_heart: |  javac  |   0m 51s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 36s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 36s |  the patch passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1  |
| +1 :green_heart: |  javadoc  |   0m 39s |  the patch passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06  |
| +1 :green_heart: |  findbugs  |   2m 33s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 21s |  tez-api in the patch passed.  |
| +1 :green_heart: |  unit  |   5m 10s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 24s |  The patch does not generate ASF License warnings.  |
|  |   |  44m 35s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.44 ServerAPI=1.44 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-356/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/356 |
| JIRA Issue | TEZ-4564 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux d50217de0167 5.15.0-94-generic #104-Ubuntu SMP Tue Jan 9 15:25:40 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / a1fcddb8b |
| Default Java | Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06 |
| mvninstall | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-356/2/artifact/out/branch-mvninstall-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-356/2/testReport/ |
| Max. process+thread count | 418 (vs. ulimit of 5500) |
| modules | C: tez-api tez-dag U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-356/2/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by tez-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 36s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   6m  3s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  12m 29s |  master passed  |
| +1 :green_heart: |  compile  |   1m 19s |  master passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1  |
| +1 :green_heart: |  compile  |   1m 11s |  master passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06  |
| +1 :green_heart: |  checkstyle  |   1m 22s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m 11s |  master passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1  |
| +1 :green_heart: |  javadoc  |   0m 59s |  master passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06  |
| +0 :ok: |  spotbugs  |   1m 18s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   2m 54s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 49s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 55s |  the patch passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1  |
| +1 :green_heart: |  javac  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 46s |  the patch passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06  |
| +1 :green_heart: |  javac  |   0m 46s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 34s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 34s |  the patch passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1  |
| +1 :green_heart: |  javadoc  |   0m 35s |  the patch passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06  |
| +1 :green_heart: |  findbugs  |   2m 18s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 15s |  tez-api in the patch passed.  |
| +1 :green_heart: |  unit  |   5m  0s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 24s |  The patch does not generate ASF License warnings.  |
|  |   |  43m 41s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.44 ServerAPI=1.44 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-356/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/356 |
| JIRA Issue | TEZ-4564 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux 48b94daddc0b 5.15.0-94-generic #104-Ubuntu SMP Tue Jan 9 15:25:40 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / a1fcddb8b |
| Default Java | Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06 |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-356/3/testReport/ |
| Max. process+thread count | 489 (vs. ulimit of 5500) |
| modules | C: tez-api tez-dag U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-356/3/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by abstractdog** [comment]
@ayushtkn , @jteagles : could you take a look at this one?

File: tez-api/src/main/java/org/apache/tez/client/TezClient.java [hunk 1]
```diff
@@ -1286,4 +1286,12 @@ public static ApplicationId appIdfromString(String appIdStr) {
...
     }
   }
+
+  public String getAmHost() {
+    return frameworkClient.getAmHost();
+  }
+
+  public int getAmPort() {
+    return frameworkClient.getAmPort();
+  }
```
Comment (ayushtkn) [inline tez-api/src/main/java/org/apache/tez/client/TezClient.java:1296]: If anyone calls these methods before ``start()`` is invoked in that case ``frameworkClient`` would be `null` right? So we need to check for `null`?

File: tez-dag/src/main/java/org/apache/tez/client/LocalClient.java [hunk 2]
```diff
@@ -204,6 +206,9 @@ public ApplicationReport getApplicationReport(ApplicationId appId) {
     report.setProgress(dagAppMaster.getProgress());
     report.setAMRMToken(null);
 
+    this.amHost = dagAppMaster.getAppNMHost();
+    this.amPort = dagAppMaster.getRpcPort();
```
Comment (ayushtkn) [inline tez-dag/src/main/java/org/apache/tez/client/LocalClient.java:210]: shouldn't this be ``getAppNMPort()``?

File: tez-api/src/main/java/org/apache/tez/client/TezClient.java (hunk 1 above)
Comment (abstractdog) [reply tez-api/src/main/java/org/apache/tez/client/TezClient.java:1296]: right, I'll make this change
a returned null value can mean that frameworkClient is not initialized at all, or initialized but hasn't connected to the AM, from the client's point of view, those are the same

**Comment by tez-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  26m 42s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   5m 45s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  11m 45s |  master passed  |
| +1 :green_heart: |  compile  |   1m 18s |  master passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1  |
| +1 :green_heart: |  compile  |   1m 10s |  master passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06  |
| +1 :green_heart: |  checkstyle  |   1m 22s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m 11s |  master passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1  |
| +1 :green_heart: |  javadoc  |   0m 59s |  master passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06  |
| +0 :ok: |  spotbugs  |   1m 18s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   2m 51s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 50s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 55s |  the patch passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1  |
| +1 :green_heart: |  javac  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 47s |  the patch passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06  |
| +1 :green_heart: |  javac  |   0m 47s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 32s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 34s |  the patch passed with JDK Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1  |
| +1 :green_heart: |  javadoc  |   0m 35s |  the patch passed with JDK Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06  |
| +1 :green_heart: |  findbugs  |   2m 16s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 14s |  tez-api in the patch passed.  |
| +1 :green_heart: |  unit  |   4m 57s |  tez-dag in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 24s |  The patch does not generate ASF License warnings.  |
|  |   |  68m 32s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.44 ServerAPI=1.44 base: https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-356/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/tez/pull/356 |
| JIRA Issue | TEZ-4564 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs checkstyle compile |
| uname | Linux b21fb11c36e6 5.15.0-94-generic #104-Ubuntu SMP Tue Jan 9 15:25:40 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/tez.sh |
| git revision | master / 30e9cdbff |
| Default Java | Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.22+7-post-Ubuntu-0ubuntu222.04.1 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_402-8u402-ga-2ubuntu1~22.04-b06 |
|  Test Results | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-356/4/testReport/ |
| Max. process+thread count | 425 (vs. ulimit of 5500) |
| modules | C: tez-api tez-dag U: . |
| Console output | https://ci-hadoop.apache.org/job/tez-multibranch/job/PR-356/4/console |
| versions | git=2.34.1 maven=3.6.3 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by ayushtkn — APPROVED** [review]
LGTM

**Pull request merged**